### PR TITLE
Chore/issues/54 add xgboost regressor model option to training evaluation pipeline

### LINKED ISSUE #54: Add XGBoost Regressor model option to training & evaluation pipeline
## Problem
FinSight-AI currently supports baseline, linear (Ridge), and a tree-based model via `HistGradientBoostingRegressor`. For stock price/return forecasting on tickers like AAPL/JPM/XOM/KO/TSLA, we want a stronger, widely-used gradient boosting baseline that typically performs well on engineered tabular features.

Add an **XGBoost** model option to the existing training/evaluation pipeline so we can compare it against the current models and use it for forecasting.

---

### In scope
- Add a new model adapter (or extend the existing tree adapter) to support **XGBoost regression** using the same interface as existing models (`ModelPort.evaluate(...)`).
- Register the model so it can be selected by `model_type` via the existing `SklearnModelRouter`.
- Ensure the model:
  - Trains using the existing feature dataset from `PandasFeatureStore` (numeric feature columns, excluding `target_column` and id columns like `date`, `ticker`).
  - Produces predictions in the same `ModelEvaluationResult` format (metrics + predictions dataframe + trained artifact).
  - Exposes feature importance in `model_metadata` (using native XGBoost importance when available; otherwise fallback to permutation importance consistent with current behavior).
- Add/update dependency management (e.g., `xgboost` in requirements / pyproject / lockfile depending on the repo setup).
- Add a minimal test or sanity check to ensure the adapter can fit/predict on a small dataset.

### Out of scope
- Hyperparameter tuning / AutoML.
- GPU training support.
- Model explainability beyond feature importance.
- Deep learning models (LSTM) or sequence models.
- Changes to feature engineering (only consume existing features).

---

## Acceptance Criteria
- [ ] `pip install` / project install succeeds with `xgboost` included as a dependency.
- [ ] A new `model_type` (e.g., `xgboost`) is available and routed through `SklearnModelRouter`.
- [ ] Running the training workflow/use case (`TrainModel`) evaluates XGBoost alongside existing models without breaking current behavior.
- [ ] `ModelEvaluationResult` returned by XGBoost adapter includes:
  - [ ] `metrics` computed via existing `forecast_metrics`
  - [ ] `predictions` dataframe including id columns (when present) + `y_true` + `y_pred`
  - [ ] `trained_artifact` set to the fitted XGBoost estimator
  - [ ] `model_metadata` populated with estimator name, hyperparameters, feature columns, and feature importance.
- [ ] Feature importance is present in `model_metadata`:
  - [ ] Prefer XGBoost native importance when available
  - [ ] Otherwise fallback to permutation importance
- [ ] Basic automated test (or equivalent check) verifies XGBoost adapter can:
  - [ ] fit on a small DataFrame with numeric features
  - [ ] predict with expected output shape
  - [ ] return a non-empty `ModelEvaluationResult`

---

## Notes / Implementation Hints
- Likely location for new adapter: `src/finsight/infrastructure/ml/sklearn/`
- Existing patterns to follow:
  - `LinearSklearnModel` (for pipeline + metadata pattern)
  - `HistGradientBoostingModel` (for feature selection + permutation importance fallback)
- Suggested baseline params (initial; tune later):
  - `n_estimators`: 300
  - `learning_rate`: 0.03
  - `max_depth`: 2-4
  - `subsample`: 0.8
  - `colsample_bytree`: 0.8
  - `reg_lambda`: 1.0
  - `random_state`: 42

---

## Definition of Done
- Code merged and model is selectable/configurable via `model_type` without regressions in existing models.

### PR DESCRIPTION
**Pull request by KTomy7**
This pull request adds support for XGBoost regression models to the machine learning infrastructure, making XGBoost available as a model option for training and evaluation. It introduces a new `XGBoostModel` adapter, integrates it into the model registry and router, updates configuration and dependencies, and provides comprehensive unit tests to ensure correct functionality and robustness.

**XGBoost Model Integration:**

* Added a new `XGBoostModel` adapter in `src/finsight/infrastructure/ml/sklearn/xgboost.py`, implementing the required interface for model evaluation, feature importance extraction, and metadata reporting.
* Registered `XGBoostModel` in the ML model router and container, enabling it for use in the application. 
* Declared XGBoost as a supported model type in configuration (`config/config.yaml`) and ensured it is included in the router's supported types.

**Dependency and Configuration Updates:**

* Added `xgboost>=2.0.0` to the project dependencies in `pyproject.toml`.

**Testing and Validation:**

* Introduced a new unit test suite for `XGBoostModel` covering fitting, prediction, metrics, feature importance, metadata, error handling, and edge cases in `tests/unit/infrastructure/ml/sklearn/test_xgboost.py`.
* Updated existing tests to include XGBoost in the list of supported model types.

**Minor Fixes:**

* Cleaned up unused imports in integration tests.
* Minor refactor in `HistGradientBoostingModel` to simplify feature importance and metadata reporting.

### CODE REVIEW

File: src/finsight/infrastructure/ml/sklearn/xgboost.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    @staticmethod
+    def _require_dataframe(dataset: object, *, arg_name: str) -> pd.DataFrame:
+        if not isinstance(dataset, pd.DataFrame):
+            raise TypeError(f"{arg_name} must be a pandas DataFrame.")
+        return dataset
+
+    @staticmethod
+    def _feature_importance_ranking(
+        feature_columns: list[str], importances: object
+    ) -> list[dict[str, float | str]]:
```
[the remark this answers is not in the document]
Comment (KTomy7) [reply src/finsight/infrastructure/ml/sklearn/xgboost.py]: Fixed it
